src/native/containers/dn-simdhash.c

```diff
@@ -0,0 +1,157 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "dn-simdhash.h"
+
+#if defined(__clang__) || defined (__GNUC__)
+static DN_FORCEINLINE(uint32_t)
+next_power_of_two (uint32_t value) {
+	if (value < 2)
+		return 1;
+	return 1u << (32 - __builtin_clz (value - 1));
+}
+#else // __clang__ || __GNUC__
+static DN_FORCEINLINE(uint32_t)
+next_power_of_two (uint32_t value) {
+	if (value < 2)
+		return 1;
+	value--;
+	value |= value >> 1;
+	value |= value >> 2;
+	value |= value >> 4;
+	value |= value >> 8;
+	value |= value >> 16;
+	value++;
+	return value;
+}
+#endif // __clang__ || __GNUC__
+
+dn_simdhash_t *
+dn_simdhash_new_internal (dn_simdhash_meta_t *meta, dn_simdhash_vtable_t vtable, uint32_t capacity, dn_allocator_t *allocator)
+{
+	const size_t size = sizeof(dn_simdhash_t) + meta->data_size;
+	dn_simdhash_t *result = (dn_simdhash_t *)dn_allocator_alloc(allocator, size);
+	memset(result, 0, size);
+
+	assert(meta);
```

Comment:
Suggestions on how to get an equivalent of g_assert here?

Reply:
> Suggestions on how to get an equivalent of g_assert here?

that's surprisingly difficult because g_assert is android/darwin aware so that it can route messages into the right system log.  Also CoreCLR in general does not keep asserts in retail builds and there's not a straightforward mechanism for a fail fast abort (that I could find, anyway).